lib/rust/ensogl/core/src/display/object/class.rs

```diff
@@ -908,6 +1120,50 @@ pub trait ObjectOps<Host = Scene>: Object<Host> {
     }
 
 
+    // === Events ===
+
+    /// Emit a new event. See docs of [`event::Event`] to learn more.
+    fn emit_event<T>(&self, event: T)
+    where
+        Host: 'static,
+        T: 'static, {
+        self.display_object()._emit_event(event)
+    }
+
+    /// Create a new event handler for the given event type. See docs of [`event::Event`] to learn
+    /// more.
+    fn on_event<T>(&self) -> frp::Source<event::Event<Host, T>>
+    where
+        Host: 'static,
+        T: frp::Data, {
+        self.display_object().rc.on_event()
+    }
+
+
+    // === Focus ===
+
+    /// Check whether this object is focused.
+    fn is_focused(&self) -> bool {
+        self.display_object()._is_focused()
+    }
+
+    /// Focus this object. See docs of [`Event::Focus`] to learn more.
+    fn focus(&self) {
+        self.display_object()._focus()
+    }
+
+    /// Blur ("unfocus") this object. See docs of [`Event::Blur`] to learn more.
+    fn blur(&self) {
+        self.display_object()._blur()
+    }
+
+    /// Blur ("unfocus") the currently focused object if any. See docs of [`Event::Blur`] to learn
+    /// more.
+    fn blur_all(&self) {
+        self.display_object()._blur_all()
+    }
```

Comment:
I miss some more explicit statement about the range of this method. From implementation, I know that it blur focused event in the tree the object belongs to (but not in other trees), but I cannot guess it from the docs.

Reply:
Good catch. I changed the name to `blur_tree` and added the following doc:

```
/// Blur the display object tree this object belongs to. If any tree node (any node directly or
    /// indirectly connected with each other) was focused, it will be blurred.
```